### PR TITLE
Improve Vale linting and reporting

### PR DESCRIPTION
**Pull request by amotl**
Hi again,

this improves the runtime of Vale on a whole filesystem tree, see #41. It might also help with #4.

- Invoking Vale just once tremendously improves efficiency
- Postprocessing now produces the artefacts "report.json", "summary.json" and "summary.csv"

With kind regards,
Andreas.

#### `summary.json`
```json
[
  {
    "file": "../CHANGES.rst",
    "errors": 1,
    "warnings": 7,
    "suggestions": 0
  },
  {
    "file": "../CODE_OF_CONDUCT.rst",
    "errors": 0,
    "warnings": 8,
    "suggestions": 0
  },
  {
    "file": "../DEVELOP.rst",
    "errors": 0,
    "warnings": 3,
    "suggestions": 0
  },
  {
    "file": "../README.rst",
    "errors": 1,
    "warnings": 1,
    "suggestions": 0
  }
]
```

#### `summary.csv`
```csv
"file","errors","warnings","suggestions"
"../CHANGES.rst",1,7,0
"../CODE_OF_CONDUCT.rst",0,8,0
"../DEVELOP.rst",0,3,0
"../README.rst",1,1,0
```

### CODE REVIEW

**Review by mechanomi — CHANGES_REQUESTED** [review]
lgtm, with one caveat

while we're at it, let's remove Vale from the `make dev` target. it was a neat idea, but for a long time now, I have felt irritated with the way it works. I just want to run `make dev`, I don't want to wait for it to lint the entire tree beforehand

additionally, you removed `bin/lint`, which is used by the build system to lint files on the fly when they are changed. so this fits in nicely

to do this, and also maybe to wrap your head around what I am requesting, just looking at the source, I would suggest the following:

- remove the `lint-watch` target entirely

- replace the entire `dev` target with this:

  ```
  .PHONY: dev
  dev: autobuild
  ```

  this effectively makes `dev` an alias for `autobuild`. all the other stuff you see there is specifically to get the on-the-fly linting working

**Comment by amotl** [comment]
@mechanomi: I added the changes you requested. Also, `lint-summary` will now be only invoked if `jq` is installed on the machine. Otherwise, the program tells the user gracefully about it without failing the target.

File: src/rules.mk [hunk 1]
```diff
@@ -83,10 +80,8 @@ help:
 	@ echo
 	@ printf 'Run `make <TARGET>`, where <TARGET> is one of:\n'
 	@ echo
-	@ printf '\033[37m  dev    \033[00m Run a Sphinx development server that'
-	@ printf                          ' builds and lints the \n'
-	@ printf '\033[37m         \033[00m documentation as you edit the source'
-	@ printf                          ' files\n'
+	@ printf '\033[37m  dev    \033[00m Run a Sphinx development server that builds\n'
```
Comment (mechanomi) [inline src/rules.mk:83]: good catch!

**Review by mechanomi — APPROVED** [review]
looks great!

before you merge, can you add an entry to CHANGES as this impacts user-facing functionality. ty!

**Comment by amotl** [comment]
I've updated the change log. Want to have a look again?

**Review by mechanomi — APPROVED** [review]

**Pull request merged**